OpenRA.Mods.Common/Traits/KillsSelfAfterDelay.cs

```diff
@@ -0,0 +1,65 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Kills actor after a certain amount of time.")]
+	public class KillsSelfAfterDelayInfo : ConditionalTraitInfo
+	{
+		[Desc("Remove the actor from the world (and destroy it) instead of killing it.")]
+		public readonly bool RemoveInstead = false;
+
+		[Desc("The amount of time (in ticks) before the actor dies. Two values indicate a range between which a random value is chosen.")]
+		public readonly int[] Delay = { 250 };
```

Comment:
We're doing ranges in quite a few places now, so I wonder whether we should introduce (not in this pr!) an `IntegerRange` type to reduce duplication and formalize (with a lint check on the values) this behaviour.